packages/anasazi/tpetra/test/IRTR/cxx_main_auxtest.cpp

```diff
@@ -0,0 +1,331 @@
+// @HEADER
+// *****************************************************************************
+//                 Anasazi: Block Eigensolvers Package
+//
+// Copyright 2004 NTESS and the Anasazi contributors.
+// SPDX-License-Identifier: BSD-3-Clause
+// *****************************************************************************
+// @HEADER
+//
+// This test is for RTR solving a generalized (Ax=Bxl) real Hermitian
+// eigenvalue problem, using the RTRSolMgr solver manager.
+//
+// This test checks that the auxiliary vector functionality, intended to support
+// checkpointing, work as intended. The test will create an eigenproblem and solve
+// for 15 eigenpairs. Then it will solve the same problem for 15 eigenpairs via checkpointing
+// three groups of 5.
+//
+#include "AnasaziConfigDefs.hpp"
+#include "AnasaziTypes.hpp"
+
+#include "AnasaziTpetraAdapter.hpp"
+#include "AnasaziBasicEigenproblem.hpp"
+#include "AnasaziRTRSolMgr.hpp"
+#include <Teuchos_CommandLineProcessor.hpp>
+#include <Teuchos_StandardCatchMacros.hpp>
+
+#include <Tpetra_Core.hpp>
+#include <Tpetra_CrsMatrix.hpp>
+
+#include <ModeLaplace1DQ1.hpp>
+
+using namespace Teuchos;
+
+
+class get_out : public std::logic_error {
+  public: get_out(const std::string &whatarg) : std::logic_error(whatarg) {}
+};
+
+
+int main(int argc, char *argv[]) 
+{
+  bool boolret;
+
+  Tpetra::ScopeGuard tpetraScope (&argc,&argv);
+  RCP<const Teuchos::Comm<int> > comm = Tpetra::getDefaultComm ();
+  int MyPID = comm->getRank();
+
+  bool testFailed = false;
+  bool verbose = false;
+  bool debug = false;
+  std::string which("LR");
+  bool skinny = true;
+
+  bool success = true;
+  try {
+
+    CommandLineProcessor cmdp(false,true);
+    cmdp.setOption("verbose","quiet",&verbose,"Print messages and results.");
+    cmdp.setOption("debug","nodebug",&debug,"Print debugging information.");
+    cmdp.setOption("skinny","hefty",&skinny,"Use a skinny (low-mem) or hefty (higher-mem) implementation of IRTR.");
+    cmdp.setOption("sort",&which,"Targetted eigenvalues (SR or LR).");
+    if (cmdp.parse(argc,argv) != CommandLineProcessor::PARSE_SUCCESSFUL) {
+      return -1;
+    }
+    if (debug) verbose = true;
+
+    typedef double                              ST;
```

Comment:
ditto

Reply:
Test is only meaningful for double-precision due to generalized eigenvalue problem being generated.